### PR TITLE
Add CI action that reports clang-tidy warnings as review comments.

### PR DESCRIPTION
**Pull request by hzeller**
#xls-build-gardener

### CODE REVIEW

**Comment by hzeller** [comment]
@proppy I am trying to get this `clang-tidy-pr-comments` action to work. The 403 looks like the action does not have permission to write. I suspect you'd be the best person knowing how the permission system is set up and what token to use.

(Note, currently the action is https://github.com/hzeller/clang-tidy-pr-comments which is just a fork from the original  https://github.com/platisd/clang-tidy-pr-comments for easier debugging).

**Comment by proppy** [comment]
@hzeller do you get the same error when running it on your fork? https://github.com/hzeller/clang-tidy-pr-comments 

That would help us to rules out project specific configuration issues.

File: .github/workflows/review-actions.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      run: |
+        xls/dev_tools/make-compilation-db.sh
+
+    - name: Run clang-tidy on diff
+      run: |
+        mkdir clang-tidy-result
+        git diff -U1000 "$(git merge-base HEAD review-base/${{ github.event.pull_request.base.ref }})" | xls/dev_tools/run-clang-tidy-diff.sh -p1 -path build -export-fixes clang-tidy-result/fixes.yml
+
+    - name: Run clang-tidy-pr-comments action
+      uses: hzeller/clang-tidy-pr-comments@master
```
Comment (hzeller) [inline .github/workflows/review-actions.yml:53]: Note, I am using my fork here.

Comment (proppy) [reply .github/workflows/review-actions.yml:53]: Sorry I meant: when using the action on your _xls_ fork (not a fork of the action).

Comment (hzeller) [reply .github/workflows/review-actions.yml:53]: ah, I see. I did not try that as my personal github does not have access to hi-umpf runners (just building the compilation DB takes more than 30min before I gave up)

**Pull request closed without merging**